include/unifex/bulk_schedule.hpp

```diff
@@ -48,22 +55,53 @@ class _schedule_receiver<Integral, Receiver>::type {
         noexcept(is_nothrow_value_receiver_v<Receiver> &&
                  is_nothrow_next_receiver_v<Receiver, Integral>) {
         using policy_t = decltype(get_execution_policy(receiver_));
-        if constexpr (is_one_of_v<policy_t, unsequenced_policy, parallel_unsequenced_policy>) {
-            // Vectorisable version
+        auto stop_token = get_stop_token(receiver_);
+        const bool stop_possible = !is_stop_never_possible_v<decltype(stop_token)> && stop_token.stop_possible();
+
+        if(stop_possible) {
+            for (Integral chunk_start(0); chunk_start < count_; chunk_start += bulk_cancellation_chunk_size) {
+                if(stop_token.stop_requested()) {
+                    unifex::set_done(std::move(receiver_));
+                    return;
+                }
+                Integral chunk_end = std::min(chunk_start + bulk_cancellation_chunk_size, count_);
```

Comment:
Have you verified whether this computation here still allows the below loop to be unrolled and vectorised?

I wonder if we need to have separate "chunk" loops which use a constant number of iterations and a tail-loop that handles the remaining iterations?

e.g. maybe something like this?
```
const Integral chunksEnd = count_ - (count_ % bulk_cancellation_chunk_size);
Integral i(0);
while (i < chunksEnd) {
  if (stopToken.stop_requested()) {
    unifex::set_done(std::move(receiver_));
    return;
  }

  const auto chunkEnd = i + bulk_cancellation_chunk_size;
  if constexpr (isVectorisable) {
    // pragma(ivdep)
    for (; i < chunkEnd; ++i) {
      unifex::set_next(receiver_, Integral(i));
    }
  } else {
    // no pragmas
    for (; i < chunkEnd; ++i) {
      unifex::set_next(receiver_, Integral(i));
    }
  }
}

for (; i < count_; ++i) {
  unifex::set_next(receiver_, Integral(i));
}
```

Reply:
I have not, largely because I would probably lean towards only including the simple version of the code here without the optimisation, until we see production use cases that benefit. I would hope that the unrolling for vectorisation you describe though is the first thing the compiler tries - the original loop would have suffered this way as well.